csrc/multidevice/utils.cpp

```diff
@@ -325,15 +331,15 @@ void propagateShardings(Fusion* fusion) {
     auto outputs = ir_utils::filterByType<TensorView>(expr->outputs());
     TensorView* input_with_mesh = nullptr;
     for (auto tv : inputs) {
-      if (tv->hasDeviceMesh()) {
+      NVF_CHECK(
```

Comment:
(Not related to this PR) cleaning up the check for propagate shardings since shardings are propagated from input to output, this just checks inputs have/don't have a sharding (by checking the DeviceMesh is set)